cps/main.c

```diff
@@ -1171,13 +1184,14 @@ run_cps(struct net_config *net_conf, struct gk_config *gk_conf,
 		cps_conf->log_ratelimit_interval_ms,
 		cps_conf->log_ratelimit_burst);
 
-	/* Take the packets needed in the KNI into account as well. */
-	front_inc = 2 * cps_conf->front_max_pkt_burst;
+	front_inc = cps_conf->front_max_pkt_burst;
 	net_conf->front.total_pkt_burst += front_inc;
 	if (net_conf->back_iface_enabled) {
-		back_inc = 2 * cps_conf->back_max_pkt_burst;
+		back_inc = cps_conf->back_max_pkt_burst;
 		net_conf->back.total_pkt_burst += back_inc;
 	}
+	/* Take the packets needed in the KNI into account as well. */
+	cps_conf->total_pkt_burst = 2 * (front_inc + back_inc);
 
```

Comment:
Account for the KNI packets is better done in `assign_cps_queue_ids()` since that, in all blocks, we only account for hardware-supported packets in the run function.